src/main/kotlin/app/RepoHasher.kt

```diff
@@ -115,11 +115,8 @@ class RepoHasher(val localRepo: LocalRepo) {
         addedCommits = mutableListOf()
 
         val commitsObservable = getObservableRevCommits()
-        commitsObservable.map({ Commit(it) }).filter {
-            val email = it.author.email
-            localRepo.hashAllAuthors || (email == localRepo.author.email ||
-                    repo.emails.contains(email))
-        }.blockingSubscribe({  // OnNext
+        commitsObservable.map(commitMapper).filter(emailFilter)
```

Comment:
Guys, let's not worry too much about patterns. At this stage, we need code that works not the code that will last 100 years. We will rewrite it once we understand what market wants.